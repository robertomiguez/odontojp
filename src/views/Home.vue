<template>
  <v-container>
    <v-layout
      text-xs-center
      wrap
    >

      <v-flex xs12>
        <div :class="[{'mb-1' : $vuetify.breakpoint.xs, 'mb-4' : $vuetify.breakpoint.smAndUp}, 'display-2 font-italic primary--text']">
          {{ title }}
        </div>
        <div :class="[{'mb-1' : $vuetify.breakpoint.xs, 'mb-4' : $vuetify.breakpoint.smAndUp}, 'display-1 font-weight-thin primary--text']">
          {{ subtitle }}
        </div>

          <div :class="{'mb-1' : $vuetify.breakpoint.xs, 'mb-4' : $vuetify.breakpoint.smAndUp}">
          <v-avatar
            :size="`${getAvatarSize()}`"
            color="grey lighten-4"
          >
            <v-img
              :src="require('@/assets/jpimage1.jpeg')"
            />
          </v-avatar>
        </div>

        <div>
          <v-btn to="/contato" round color="primary" dark>Contato</v-btn>
        </div>
      </v-flex>

      <v-flex xs12>
      </v-flex>
  </v-layout>

  </v-container>
</template>

<script>
export default {
  data: () => ({
    title: 'Bemvindo ao Consultório!',
    subtitle: 'Dr. João Paulo'
  }),
  methods: {
    getAvatarSize () {
      if (this.$vuetify.breakpoint.xs)
        return '250'
      else
        return '400'
    }
  }
}
</script>

<style>
</style>
