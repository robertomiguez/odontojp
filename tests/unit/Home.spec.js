import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import Home from '@/views/Home'

Vue.use(Vuetify)

describe('Home', () => {
  it('renders the component', () => {
    // arrange
    const wrapper = shallowMount(Home)
    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Checking DIV subtitle tag text', () =>{
    const wrapper = shallowMount(Home)
    const div = wrapper.find('div')
    expect(div.text()).toBe('Bemvindo ao Consult√≥rio!')
  })
})
