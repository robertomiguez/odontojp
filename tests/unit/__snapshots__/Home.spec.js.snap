// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the component 1`] = `
<v-container-stub tag="div">
  <v-layout-stub tag="div" text-xs-center="" wrap="">
    <v-flex-stub tag="div" xs12="">
      <div class="mb-4 display-2 font-italic primary--text">
        Bemvindo ao Consultório!
      </div>
      <div class="mb-4 display-1 font-weight-thin primary--text">
        Dr. João Paulo
      </div>
      <div class="mb-4">
        <v-avatar-stub color="grey lighten-4" size="250">
          <v-img-stub src="" position="center center" transition="fade-transition"></v-img-stub>
        </v-avatar-stub>
      </div>
      <div>
        <v-btn-stub color="primary" activeclass="v-btn--active" to="/contato" tag="button" dark="true" round="true" type="button">Contato</v-btn-stub>
      </div>
    </v-flex-stub>
    <v-flex-stub tag="div" xs12=""></v-flex-stub>
  </v-layout-stub>
</v-container-stub>
`;
